### ISSUE TITLE
F2 Score metric for object detection.

### ISSUE THREAD

**Issue by innat** [keras-team/keras-cv#393]
The [F2-score](https://en.wikipedia.org/wiki/F-score) was used to evaluate the performance in [**TensorFlow - Help Protect the Great Barrier Reef**](https://www.kaggle.com/competitions/tensorflow-great-barrier-reef/overview/evaluation). It might be a good addition in keras-cv.

**Comment by bhack** [comment]
https://github.com/tensorflow/addons/blob/master/tensorflow_addons/metrics/f_scores.py

**Comment by quantumalaviya** [comment]
Wouldn't it be better to just write a general FScore metric and change the beta values?

**Comment by bhack** [comment]
> F2Score class that just subclasses FScore and provides a beta of 2.
...I think this is like my previous comment

**Comment by markub3327** [comment]
FBetaScore should be accept the sparsely defined labels.

For example:
```python
# check for sparse labels
if tf.rank(y_true) < tf.rank(y_pred):
    y_true = tf.one_hot(y_true, self.num_classes)
```

**Comment by markub3327** [comment]
@bhack 

`tensorflow_addons/metrics/f_scores.py` to `keras_cv/metrics/f_scores.py` and my PR too?

**Comment by bhack** [comment]
Yes and if you want also F2 with the related tests. 
Note, that you could need or not to adapt/refactor these a bit for this repo standards, we have few metrics integrated at https://github.com/keras-team/keras-cv/tree/master/keras_cv/metrics/coco

**Comment by markub3327** [comment]
@bhack Okey, let's do it.

**Comment by bhack** [comment]
> @innat Or, it can be argument:
> 
> `sparse=True` or `sparse=False`

@markub3327 Check the thread at https://github.com/keras-team/keras-cv/pull/371#discussion_r866306775

**Comment by markub3327** [comment]
@bhack Ok

**Comment by Suzan009** [comment]
Any update on this? 
Already asked. https://github.com/keras-team/keras/issues/15811

**Comment by tanzhenyu** [comment]
I think we should add F1 score to Keras, and keep F2 score to KerasCV?

**Comment by tanmaypandey7** [comment]
@tanzhenyu why don't we keep both f1 and f2 scores to keras itself? Won't it be a bit confusing to the user, if he needs to import the f1 metric from keras and f2 from keras-cv?

Also, i would like to work/help on this PR if there is nobody working on it currently. 
cc: @markub3327

**Comment by innat** [comment]
https://github.com/keras-team/keras/pull/17516

**Comment by bhack** [comment]
https://github.com/keras-team/keras-cv/pull/423

**Comment by LukeWood** [comment]
closing as this most likely fits in keras core

**Comment by innat** [comment]
@LukeWood 
Could u please move this ticket to keras?

**Comment by LukeWood** [comment]
> @LukeWood Could u please move this ticket to keras?

I do not have Keras ownership O_o.  Can we re-open there with a link here?
